### DISCUSSION TITLE
Any plans on supporting swapping spaces between displays?

### DISCUSSION THREAD

**Discussion by Subjective** [asmvik/yabai#2103] [Q&A] [answered]
Currently, it's possible to move spaces back and forth between displays with `yabai -m space --display recent`. However, this sets original display to the first space, so the relative ordering of spaces is not preserved when moving a space back to the original display.

**Comment by asmvik** [comment, marked as answer]
Implemented on master; https://github.com/koekeishiya/yabai/issues/2114

yabai will attempt to focus the previous space, when moving the active space to another display.
Calling `yabai -m space --display recent` repeatedly should now work fine.

e.g: 
display 1 has space 1 and 2, with focus on space2.
display 2 has space 3 and 4, with focus on space 3.

if you are on display 1, space 2, and run `yabai -m space --display recent`
display 1 will focus space 1, and space 2 will be moved to display 2 after space 3. 

the new state will be:
display 1 has space 1 with focus on space 1.
display 2 has space 2, 3 and 4, with focus on space 3.

note that display 2 space 3 was originally space 2 on display 1, where as display 2 space 2 and 4 are the "original space 3 and 4".
